### PR TITLE
fix(gfql): preserve boolean sum zero on polars gpu

### PR DESCRIPTION
**Pull request by lmeyerov**
## Summary

- Normalize the cuDF-Polars 26.02 all-null Boolean `sum` result to the GFQL/Cypher zero contract.
- Scope the repair to exactly Boolean `sum` before the existing Int64 result cast.
- Add direct positive and negative boundary tests.
- Record the fix under `[Development]` in `CHANGELOG.md`.

## Final linear state

- Head: `d3573735040d7f65d00a24f2e08fa5b8a9d83e77`
- Base: `master` at merged #2001 commit `d627fac2e7e55168f0ba43b8b1f4e3311b4e2f52`
- Topology: five linear, one-parent commits; zero merge commits
- Incremental diff: three files, +24/-0

The four previously reviewed code/test patches are range-diff exact after the final rebase. The fifth commit adds only the requested changelog entry.

## Semantics boundary

`polars_conform_agg_dtype()` fills null only when both conditions hold: the aggregate is `sum` and the input dtype is Boolean.

The focused test pins:

- Positive: Boolean `sum` null → Int64 zero.
- Negative function side: Boolean `count` null → Int64 null.
- Negative dtype side: non-Boolean `sum` null → unchanged null.

## Validation

- Focused CPU boundary selection: 52 passed, 18 expected skips, 425 deselected.
- `./bin/lint.sh`: clean, including type-hygiene, comment-density, and import guards.
- `./bin/typecheck.sh`: mypy 2.3.1, 336 source files, zero issues.
- No Docker, local GPU, DGX, or broad local suite was used for the final rebase.

## Review resolution

- [Boundary test request](https://github.com/graphistry/pygraphistry/pull/2002#discussion_r3833384063): resolved by the focused positive cell and two negative controls.
- The branch is now directly linear on merged #2001 with no integration merge or predecessor diff leakage.

## Landing

Review the master-relative range `d627fac2e...d35737350`.

After this exact-head CI generation is green, the authorized landing method is GitHub **Create a merge commit** with admin override. Retain the source branch.

### CODE REVIEW

**Comment by lmeyerov** [comment]
Stacked incorrectly

File: graphistry/compute/gfql/agg_types.py [hunk 1]
```diff
@@ -129,9 +129,13 @@ def polars_agg_result_cast(func: str, input_dtype: "Optional[pl.DataType]") -> "
 def polars_conform_agg_dtype(expr: "pl.Expr", func: str, input_dtype: "Optional[pl.DataType]",
                              alias: str) -> "pl.Expr":
     """Land a polars aggregate on its CONTRACT dtype rather than on its kernel dtype."""
+    import polars as pl
+
     target = polars_agg_result_cast(func, input_dtype)
     if target is None:
         return expr.alias(alias)
+    if func == "sum" and input_dtype == pl.Boolean:
+        expr = expr.fill_null(0)
```
Comment (lmeyerov) [inline graphistry/compute/gfql/agg_types.py:138]: Add positive and negative testing on either side of the boundary

**Comment by lmeyerov** [comment]
Resolved on exact head `712e2d121f6441b4cef320469a6c6162ea4a63bd`: #2002 is now four one-parent commits directly on exact #2001 head `a888c33f2d42f626bb0754fb1186b2d1c1bf6561`, with zero integration merges. The incremental diff is exactly two files (+23/-0), and all four patches are range-diff equivalent to their reviewed versions. The PR remains based on the #2001 branch for incremental review; after #2001 lands, the maintainer can retarget #2002 to `master`, wait for the refreshed exact-head checks, and merge out of band.

File: graphistry/compute/gfql/agg_types.py (hunk 1 above)
Comment (lmeyerov) [reply graphistry/compute/gfql/agg_types.py:138]: Addressed by exact commit `712e2d121f6441b4cef320469a6c6162ea4a63bd`. The new focused helper test pins the positive cell (Boolean `sum` null -> Int64 zero) and both negative conjuncts (Boolean `count` null stays Int64 null; non-Boolean `sum` null stays unchanged null). Post-rebase focused CPU validation is 21 passed / 20 deselected; lint and mypy are clean, with zero added prohibited typing or dynamic-attribute tokens.
